File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
       #dictionary
       prev_map = {}
       #loop index i, key n in nums
       for i,n in enumerate(nums):
           #difference = target - current num
           diff = target - n
#if we have in the map the difference?
#return the number in the index of difference, what we need to find.
#and his index in the list of nums
           if diff in prev_map:
                return[prev_map[diff],i]
           prev_map[n] = i

       

            

File: Data Structures & Algorithms/duplicate-integer/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
       
        const countMap = new Map();

        if(nums.length<2){
            return false
        }

        for(let num of nums){
            if(countMap.has(num)){
                return true
            }
            countMap.set(num,true)
            
        }
        return false
    }
}
